app/models/user.rb

```diff
@@ -1,4 +1,29 @@
 class User < ActiveRecord::Base
   has_many :bids
   validates :email, email: true, allow_blank: true
+  before_save :clear_sam_status_if_duns_changed
```

Comment:
There is probably only one place in one controller where this duns check needs to be done. This logic can be wrapped up there, and the User can just remain data. Tests don't get funky and have side effects. And lots of other unexpected side effect averted hopefully too